lib/src/system_cache.dart

```diff
@@ -35,8 +35,14 @@ class SystemCache {
     if (Platform.environment.containsKey('PUB_CACHE')) {
       return Platform.environment['PUB_CACHE'];
     } else if (Platform.isWindows) {
+      // If a cache dir already exists in %APPDATA%, use it, else default to %LOCALAPPDATA%
       var appData = Platform.environment['APPDATA'];
-      return p.join(appData, 'Pub', 'Cache');
+      var appDataCacheDir = p.join(appData, 'Pub', 'Cache');
+      if (Directory(appDataCacheDir).existsSync()) {
```

Comment:
nit, but for consistency with the rest of the code base can we consider using [`dirExists(dir)`](https://github.com/dart-lang/pub/blob/ceaa86f2c7eb35f044b4c202268a17361de68d13/lib/src/io.dart#L303-L307).